pkg/cgroups/manager/manager.go

```diff
@@ -249,6 +270,7 @@ func (m *CgroupManager) processPodEvents() {
 			case podDumpMetadataEvent:
 				m.dumpPodMetadata(ev.allMetadataOut)
 			}
+			m.wg.Done()
 		case <-m.shutdown:
 			return
```

Comment:
nit: Incidentally, one of the concerns for holding the WaitGroup longer is that in this case there could be a chance that the shutdown signal arrives right as a podEvent is handled, so then the queued event(s) will never be handled => never call `m.wg.Done()` => Potentially block callers? Of course if this is only used for testing then it's not a problem... unless someone in future attempts to use it for other purposes, then they would be in for some surprises.